src/Components/SortingSection/SortingSection.js

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import './SortingSection.css';
+import PropTypes from 'prop-types';
+import { GoodsList } from '../GoodsList/GoodsList';
+
+class SortingSection extends React.Component {
+  state = {
+    goodsList: this.props.goods,
+    sortAlphabetically: true,
+    sortLength: true,
+    checkboxValue: 10,
+  };
+
+  handleReverse = () => {
+    this.setState(state => ({ goodsList: state.goodsList.reverse() }));
+  }
+
+  handleAlphabeticalSort = () => {
+    const { sortAlphabetically } = this.state;
+
+    this.sortAlphabeticallyForwards = () => {
+      this.setState(state => ({
+        goodsList: [...state.goodsList].sort(),
+        sortAlphabetically: !state.sortAlphabetically,
+      }));
+    };
+
+    this.sortAlphabeticallyBackwards = () => {
+      this.setState(state => ({
+        goodsList: [...state.goodsList].reverse(),
+        sortAlphabetically: !state.sortAlphabetically,
+      }));
+    };
+
+    sortAlphabetically
+      ? this.sortAlphabeticallyForwards()
+      : this.sortAlphabeticallyBackwards();
+  }
+
+  handleLengthSort = () => {
+    const { sortLength } = this.state;
+
+    this.sortLengthForwards = () => {
+      this.setState(state => ({
+        goodsList: [...state.goodsList].sort((a, b) => a.length - b.length),
+        sortLength: !state.sortLength,
+      }));
+    };
+
+    this.sortLengthBackwards = () => {
+      this.setState(state => ({
+        goodsList: [...state.goodsList].sort((a, b) => b.length - a.length),
+        sortLength: !state.sortLength,
+      }));
+    };
+
+    sortLength
+      ? this.sortLengthForwards()
+      : this.sortLengthBackwards();
+  }
+
+  handleReset = () => {
+    this.setState(state => ({ goodsList: [...this.props.goods] }));
+  }
+
+  handleSelectChange = (value) => {
+    this.setState(() => ({
+      goodsList: [...this.props.goods].filter((_, index) => index < value),
```

Comment:
Надо фильтровать по длине слов, а не индексу в массиве
```suggestion
      goodsList: this.props.goods.filter((_, index) => index < value),
```
Фильтр не мутирует исходный массив